cmake/obs-websocket-api.cmake

```diff
@@ -0,0 +1,18 @@
+add_library(obs-websocket-api INTERFACE)
+add_library(OBS::websocket-api ALIAS obs-websocket-api)
+
+target_sources(obs-websocket-api INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/lib/obs-websocket-api.h>
+                                           $<INSTALL_INTERFACE:${OBS_INCLUDE_DESTINATION}/obs-websocket-api.h>)
+
+target_link_libraries(obs-websocket-api INTERFACE OBS::libobs)
+
+target_include_directories(obs-websocket-api INTERFACE "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/lib>"
+                                                       "$<INSTALL_INTERFACE:${OBS_INCLUDE_DESTINATION}>")
+
+set_target_properties(
+  obs-websocket-api
+  PROPERTIES FOLDER plugins/obs-websocket
```

Comment:
The folder property is actually superfluous here because for INTERFACE libraries the sources are added directly into consumer's source list, they'll never appear as their own entries in MSVC' or Xcode's project view, but it also doesn't break anything.